Application/src/model/Menu.java

```diff
@@ -163,29 +163,39 @@ public static void pause() {
     private void setEvents(Stage stage) {
 
         button_import.setOnAction((ActionEvent event) -> {
-                Collection<String> result = Import.file(stage);
-                if (result.isEmpty()) return;
+            Collection<String> result = Import.file(stage);
+            if (result == null) return;
+            if (result.isEmpty()) return;
 
-                // Parse the element if it contains a supported diagram
-                Parser parse = new Parser();
+            // Parse the element if it contains a supported diagram
+            Parser parse = new Parser();
+
+            for (String file: result) {
                 switch(DiagramCheck.ContainsDiagram(result)) {
                     case "sequence_diagram" :
-                        for (String element : result) {
-                            parse.parseSequenceDiagram(element);
-                            Net.push(parse.getFirstSequenceDiagram());
-
-                            // Enable all media buttons.
-                            setMenuState(true);
-                            button_previous.setDisable(true);
-                        }
+                        parse.parseSequenceDiagram(file);
+
+                        Net.push(parse.getDiagram());
+                        Net.push(parse.getParallelSequenceDiagram());
+
+                        // Enable all media buttons.
+                        setMenuState(true);
+                        button_previous.setDisable(true);
+                        break;
+                    case "class_diagram":
+                        parse.parseClassDiagram(file);
+                        System.out.println(parse.getDiagram()); // TODO print until backend is implemented.
```

Comment:
So the user story is not done?

Reply:
@timjon the story is done, this line just shows what parsing returns to facilitate integration, i don't think #33 should be blocked